scripts/python/src/fodt/keyword_linker.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import re
+import xml.sax
+import xml.sax.handler
+import xml.sax.xmlreader
+import xml.sax.saxutils
+
+from pathlib import Path
+
+import click
+
+from fodt.constants import ClickOptions, Directories, FileNames, FileExtensions
+from fodt.exceptions import HandlerDoneException, ParsingException
+from fodt import helpers
+from fodt.xml_helpers import XMLHelper
+
+class ExtractURI_Handler(xml.sax.handler.ContentHandler):
+    def __init__(self, keyword_name: str) -> None:
+        self.keyword_name = keyword_name
+        self.in_section = False
+        self.in_bookmark = False
+        self.in_bookmark2 = False
+        self.uri = ""
+
+    def characters(self, content: str):
+        if self.in_bookmark2:
+            # Check if the content matches the keyword name
+            match =  re.search(self.keyword_name, content)
+            if match is None:
+                #self.uri = None
+                #raise ParsingException(
+                #    f"Keyword name {self.keyword_name} not found in bookmark content: {content}"
+                #)
+                # NOTE: Since the content may contain span tags, for example the CO2STORE keyword goes like this
+                #  <text:span text:style-name="T6">C</text:span>O2STORE
+                # we cannot expect to find the keyword name in the content. Instead we print a warning and
+                # the user can manually check the warning message to see if the keyword name was found.
+                logging.warning(f"Keyword name {self.keyword_name} not found in bookmark content: {content}")
+            raise HandlerDoneException("Done parsing.")
+
+    def endDocument(self):
+        raise ParsingException("Keyword name not found in document")
+
+    def endElement(self, name: str):
+        if name == "text:h":
+            if self.in_section:
+                self.in_section = False
+                raise ParsingException("Keyword name not found in document")
```

Comment:
Can you please comment and document why we raise an exception here? It is not immediately clear to me why the keyword name was not found when we throw ParsingException.

The name ParsingException seems a bit misleading. This might be personal taste, though.

Reply:
> why we raise an exception here?

It is just for documentation that this should never happen  (and also to catch an unexpected state early which will help with debugging) . If we reach this part of the code we will not have any hope of finding the URI value (since it should always be at the top of heading level 3 in the keyword file).